kurobako/src/log/logger.h

```diff
@@ -0,0 +1,32 @@
+#ifndef LOGGER_H
+#define LOGGER_H
+#include <array>
+#include "types/sizedtypes.h"
+#include "types/nonpersistentstring.h"
+#include <mutex>
+
+#define LOG_INFO(str,level) GET_SINGLETON_SCOPED(kurobako::log,Logger)->Log(kurobako::NonPersistentString("%s(%d) [INFO] > %s",__FILE__,__LINE__,str.Get()),level)
+#define LOG_ERROR(str,level) GET_SINGLETON_SCOPED(kurobako::log,Logger)->Log(kurobako::NonPersistentString("%s(%d) [ERROR] > %s",__FILE__,__LINE__,str.Get()),level)
+#define LOG_DEBUG(str,level) GET_SINGLETON_SCOPED(kurobako::log,Logger)->Log(kurobako::NonPersistentString("%s(%d) [DEBUG] > %s",__FILE__,__LINE__,str.Get()),level)
+namespace kurobako::log
+{
+    class Logger
+    {
+		enum
+		{
+			LOG_BUFFER_SIZE = 1 << 20
```

Comment:
Spaces, 4.

Will probably use power tools to convert everything to spaces -.- Visual studio keeps resetting my settings for god uinknown reasons